doc/Contribution.md

```diff
@@ -0,0 +1,106 @@
+# Contribution Guidelines
+==================
+## Overview
+Being an Open Source project, everyone can contribute, provided that it respect the following points:
+* Before contributing any code, the author must make sure all the tests work (see below how to launch the tests).
+* Developed code must adhere to the syntax guidelines enforced by the linters.
+* Code must be developed following the branching model and changelog policies defined below.
+* For any new feature added, unit tests must be provided, following the example of the ones already created.
+
+In order to start contributing:
+1. Fork this repository clicking on the "Fork" button on the upper-right area of the page.
+2. Clone your just forked repository:
+```
+git clone https://github.com/your-github-username/iotagent-node-lib.git
+```
+3. Add the main iotagent-node-lib repository as a remote to your forked repository (use any name for your remote
+repository, it does not have to be iotagent-node-lib, although we will use it in the next steps):
+```
+git remote add iotagent-node-lib https://github.com/telefonicaid/iotagent-node-lib.git
+```
+
+Before starting contributing, remember to synchronize the `develop` branch in your forked repository with the `develop`
+branch in the main iotagent-node-lib repository, by following this steps
+
+1. Change to your local `develop` branch (in case you are not in it already):
+```
+  git checkout develop
+```
+2. Fetch the remote changes:
+```
+  git fetch iotagent-node-lib
+```
+3. Merge them:
+```
+  git rebase iotagent-node-lib/develop
+```
+
+Contributions following this guidelines will be added to the `develop` branch, and released in the next version. The
+release process is explaind in the *Releasing* section below.
+
+## Branching model
+There are two special branches in the repository:
+
+* `master`: holds the code for the last stable version of the project. It is only updated when a new version is released,
+and its always updated with the current state of `develop`.
+
+* `develop`: contains the last stable development code. New features and bug fixes are always merged to `develop`.
+
+In order to start developing a new feature or refactoring, a new branch should be created with name `task/<taskName>`.
```

Comment:
In CB we use the following patterns for branches (I think that aligned with TID SCM policies), depending on the kind of work:

feature/XXX
hardening/XXX
bug/XXX

We also use task/XXX but a sub-level, e.g. a complex feature/XXX that needs several steps to be completed is worked in several PRs, each one a task/XXX -> feature/XXXX.